lib/httputils/statconn.go

```diff
@@ -0,0 +1,93 @@
+package httputils
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+	"sync/atomic"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/netutil"
+	"github.com/VictoriaMetrics/metrics"
+)
+
+var (
+	dialsTotal *metrics.Counter
+	dialErrors *metrics.Counter
+	conns      *metrics.Counter
+
+	connReadsTotal   *metrics.Counter
+	connWritesTotal  *metrics.Counter
+	connReadErrors   *metrics.Counter
+	connWriteErrors  *metrics.Counter
+	connBytesRead    *metrics.Counter
+	connBytesWritten *metrics.Counter
+)
+
+var metricSetRegister sync.Once
+
+// GetStatDialFunc returns dial function that supports DNS SRV records,
+// and register stats metrics for conns.
+func GetStatDialFunc(metricPrefix string) func(ctx context.Context, network, addr string) (net.Conn, error) {
+	metricSetRegister.Do(func() {
+		dialsTotal = metrics.NewCounter(fmt.Sprintf(`%s_dials_total`, metricPrefix))
```

Comment:
do we need custom metric name for each component?

Reply:
Why not? I think it is useful to have these metrics, unless they start to collide.

Reply:
imho finding needed data on grafana is easier, when you have a static metric name with different label values

Reply:
I think differently, when you know which metric you wanna use, having different metricName or different labels with same metricName are the same.
But when you can't remember the exact metricName, using auto-complete, having specific metricName prefix like `remotewrite` or `remoteread` can help to find metrics that component exposes for specific feature.